src/components/PdfFetcher/index.js

```diff
@@ -0,0 +1,163 @@
+    }
+  }
+
+  isRootPage (page) {
+    return page.numericId === 0
+  }
+
+  fetchUrl () {
+    const page = this.props.page
+    const url = `https://cms.integreat-app.de/${this.props.location}/wp-admin/admin-ajax.php`
+    const pageIds = []
+    const requestType = 'page' // 'allpages' is available for the root page, but 'allpages' doesn't work with all
+                               // languages, so we just always use 'page' as requestType.
+    const font = this.getFont()
+    const title = this.isRootPage(page) ? this.getLocationTitle(page) : page.title
+    const toc = isEmpty(page.children) ? 'false' : 'true'
+
+    this.setState(Object.assign({}, this.state, {loading: page}))
+
+    if (!this.isRootPage(page)) {
+      pageIds.push(page.numericId)
+    }
+    this.addPageIdsRecursively(pageIds, page)
+
+    const params = {
+      action: 'frontEndDownloadPDF',
+      requestType: requestType,
+      myContent: pageIds.join(','),
+      pdfOptions: `${toc},${title},${page.id}_file,,`,
+      'ajaxVars[ajaxurl]': `https://cms.integreat-app.de/${this.props.location}/wp-admin/admin-ajax.php`,
+      font,
+      fontcolor: '',
+      bgcolor: '',
+      linkcolor: ''
+    }
+
+    const body = new URLSearchParams(params)
+    // Currently the backend can only use the 'Referer' header, to determine which language the PDF should have (not sure).
+    // We cannot modify the 'Referer'-header and we are just lucky that it works I think.
+    // The old JQuery webapp also sended following cookies to the backend which it maybe used to determine the language.
+    // 'Cookie': `_icl_current_language=${this.props.languageCode}; integreat_lang=${this.props.languageCode}`
+    let text = ''
+    const decoder = new TextDecoder()
+    chunkedRequest({
+      url,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded;' // Currently IE11 does another request which fails because it appends a \n before Content-Type
+                                                            // But other request succeeds so let's ignore it
+      },
+      body,
+      chunkParser: (bytes) => { text += decoder.decode(bytes) },
+      onComplete: () => {
+        if (!page === this.state.loading) {
+          return
+        }
+
+        const regex = escapeRegExp(`https://cms.integreat-app.de/${this.props.location}/wp-content/uploads/`) + '[\\w|/|-]*\\.pdf'
+        const match = text.match(new RegExp(regex))
+
+        if (isEmpty(match)) {
+          console.error('No match found for PDF-Url.')
+          this.setState({loading: false})
+          return
+        }
+
+        this.setState({pdf: match[0], loading: false})
+        // PdfProcessingPage shouldn't appear in browser history
+        window.location.replace(match[0])
+      }
+    })
+  }
+
+  render () {
+    const t = this.props.t
+    if (this.state.loading) {
+      return <div className={style.pdfFetcher}>
+        <p>{t('common:creatingPdf')}</p>
```

Comment:
I think p tags should only be used for paragraphs. As this is no continious text there are no paragraphs.

Reply:
That's alright.